src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -3941,7 +3941,24 @@ module TcDeclarations =
              | SynMemberDefn.NestedType (range=m) :: _ -> errorR(Error(FSComp.SR.tcTypesCannotContainNestedTypes(), m))
              | _ -> ()
         | ds ->
-            // Classic class construction 
+            // Check for duplicated parameters
+            ds
```

Comment:
Please just use a for loop - we almost never use `List.iter` or `Array.iter` etc. in the compiler.
```fsharp
for slot in ds do
    if isAbstractSlot ds then
        match slot with ...
```
etc.

Also for `argNames` loop

Reply:
wow the code looks much simpler now . Thanks :)